Index: ccsm_utils/Case.template/config_compsets.xml
===================================================================
--- ccsm_utils/Case.template/config_compsets.xml	(revision 67498)
+++ ccsm_utils/Case.template/config_compsets.xml	(working copy)
@@ -174,6 +174,7 @@
 <COMPSET sname="B_RCP6.0_CN_GLC_CISM1"    alias="BGRCP60CN" grid="gland">RCP6_CAM4_CLM40%CN_CICE_POP2_RTM_CISM1_SWAV</COMPSET>
 <COMPSET sname="B_RCP8.5_CN_GLC_CISM1"    alias="BGRCP85CN" grid="gland">RCP8_CAM4_CLM40%CN_CICE_POP2_RTM_CISM1_SWAV</COMPSET>
 
+<COMPSET sname="B_1850_CAM5_CLM45_GLC_CISM1"     alias="BG1850C5L45"       grid="gland">1850_CAM5_CLM45%SP_CICE_POP2_RTM_CISM1_SWAV</COMPSET>
 <COMPSET sname="B_1850_CAM5_CLM45BGC_GLC_CISM2"  alias="BG1850C5L45BGCIS2" grid="gland">1850_CAM5_CLM45%BGC_CICE_POP2_RTM_CISM2_SWAV</COMPSET>
 
 <!-- C compsets -->
Index: ccsm_utils/Tools/cesm_postrun_setup
===================================================================
--- ccsm_utils/Tools/cesm_postrun_setup	(revision 67498)
+++ ccsm_utils/Tools/cesm_postrun_setup	(working copy)
@@ -22,13 +22,15 @@
   exit -1
 endif
 set exceeded_max_time = 0
-if ( `grep -q 'SUCCESSFUL TERMINATION' $CPLLogFile` != 0 ) then
+grep -q 'SUCCESSFUL TERMINATION' $CPLLogFile
+if ( $status == 0 ) then
    echo "run SUCCESSFUL $sdate" >>& $CASEROOT/CaseStatus
 else
-  if ( `grep -q 'Wall clock time exceeds max_cplstep_time' $CPLLogFile` != 0 ) then
+  grep -q 'Wall clock time exceeds max_cplstep_time' $CPLLogFile
+  if ( $status == 0 ) then
     echo "Model timestep time exceeded max_cplstep_time"
     echo "run EXCEEDED max_cplstep_time $sdate" >>& $CASEROOT/CaseStatus 
-    exceeded_max_time = 1
+    set exceeded_max_time = 1
   else
     echo "Model did not complete - see $RUNDIR/$CESMLogFile" 
     echo "run FAILED $sdate" >>& $CASEROOT/CaseStatus 
